### PR TITLE
ftintitle plugin now allows a custom format to be defined (Correct Branch)

### PR DESCRIPTION
**Pull request by amishb**
(Apologies for the duplicate request, but I referenced the incorrect branch. Thanks.)

I have updated the ftintitle plugin to allow a custom format to be defined. The default format is that which was previously hardcoded.

This pull request allows formats like the list below to be defined:

* feat. ...
* (Feat. ...)
* ft. ...
* featuring ...

This is my first commit to an open source project, so I'm still working my way around the beets project. Any feedback would be great.

Thanks

### CODE REVIEW

**Comment by brunal** [comment]
Hi,
Could you add a test in `test/test_ftintitle.py`? And set `feat.` as the default value in the default config (`beets/default_config.yaml`)?

Finally shouldn't we allow *multiple formats*: a list of formats instead of a single format?

**Comment by amishb** [comment]
I'll add the tests. How would you go about testing a ```item.store()```

I didn't add the default value because not all plugins register their defaults in the default config (ftintitle wasn't there previously, do I didn't add to it), I will add it however if required. 

In regards to into the multiple formats, why would you want to define a list. I would want my music to follow the same format of how I define my feat. I'm unsure as to the use case in which this would be relevant.

**Comment by brunal** [comment]
> I'll add the tests. How would you go about testing a item.store()

Why would you want to test it? What's important is testing the `ft_in_title()` method IMO. And if you want to do functional testing with `TestHelper.run_command()` (which is a nice additional test) you don't need to mock `Item.store()` since it happens in a temporary beets environment (see `TestHelper.setup_beets()` and its usages).
In general (but not in this case) you would mock the method: `item.store = mock.Mock`. That way you can even assert how many times it's been called and with what arguments (see usages of `assert_has_calls` and `assert_called_once`).

> I didn't add the default value because not all plugins register their defaults in the default config [...]

Good point.

> In regards to into the multiple formats, why would you want to define a list.
Never mind. I don't use that plugin and I was mistaken on its functionality.

**Comment by amishb** [comment]
Hi @brunal 

I have simplified the implementation, as the previous code was passing parameters into functions that it didn't need to. 

Thank you also for pointing me in the direction for the functional tests. I learnt a great deal about them and gave it my best shot. The ftintitle plugin has no functional tests at all, so I created a new class for them, and have added a few, not only surrounding my changes but also for the "drop" feat option too.

I hope this is okay.

**Comment by amishb** [comment]
Looks like I have some issues to sort out in the build. I'll get on it.

**Comment by amishb** [comment]
Build looks good. Comments are welcome :)

File: docs/plugins/ftintitle.rst [hunk 1]
```diff
@@ -24,6 +24,9 @@ file. The available options are:
 - **drop**: Remove featured artists entirely instead of adding them to the
   title field.
   Default: ``no``.
+- **format**: Defines the format for the feat part of the new title field.
+  In this format the ``{0}`` is used to define where the featured artists are placed
```
Comment (sampsyo) [inline docs/plugins/ftintitle.rst]: Missing a period at the end of the sentence.

File: docs/plugins/ftintitle.rst [hunk 2]
```diff
@@ -24,6 +24,9 @@ file. The available options are:
 - **drop**: Remove featured artists entirely instead of adding them to the
   title field.
   Default: ``no``.
+- **format**: Defines the format for the feat part of the new title field.
```
Comment (sampsyo) [inline docs/plugins/ftintitle.rst]: For clarity, maybe:

    for the "featuring X" part of

File: beetsplug/ftintitle.py [hunk 3]
```diff
@@ -82,6 +82,7 @@ def __init__(self):
         self.config.add({
             'auto': True,
             'drop': False,
+            'format': u'feat. {0}'
```
Comment (sampsyo) [inline beetsplug/ftintitle.py]: A comma at the ends of this line helps with maintainability. (Tiny nitpick, I know. :flushed:)

**Comment by sampsyo** [comment]
LGTM! I made a few very tiny style suggestions, after which I think this will be all ready.

And the tests are awesome!

**Comment by amishb** [comment]
Hi @sampsyo 
The changes you requested have been added.

:)

**Comment by sampsyo** [comment]
Awesome! :sparkles: Thanks for implementing this. All merged.

**Pull request merged**